### PR TITLE
Added OptionalWhenBraces rule

### PR DESCRIPTION
**Pull request by schalkms**
implements #335

### CODE REVIEW

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/OptionalWhenBracesSpec.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+					when (1) {
+						1 -> { foo() }
+						2 -> foo
+						3 -> {
+							foo()
+							bar()
+						}
+					}
+				}"""
+			Assertions.assertThat(subject.lint(code)).hasSize(1)
```
Comment (vanniktech) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/OptionalWhenBracesSpec.kt]: also here static import?

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/OptionalWhenBracesSpec.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.jetbrains.spek.api.dsl.describe
+import org.jetbrains.spek.api.dsl.it
+import org.jetbrains.spek.subject.SubjectSpek
+
+class OptionalWhenBracesSpec : SubjectSpek<OptionalWhenBraces>({
+	subject { OptionalWhenBraces() }
+
+	describe("check optional braces in when expression") {
+
+		it("has unnecessary braces") {
```
Comment (Mauin) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/OptionalWhenBracesSpec.kt]: We should also add tests that verify it doesn't report anything when no `when` is present, and when the `when` is okay

**Comment by vanniktech** [comment]
What will happen when there is a comment? So like:

```kotlin
when (5) {
  4 -> {
    // Foo bla.
    foo()
  }
}
```

I'd expect it to not get flagged

**Comment by arturbosch** [comment]
Good point @vanniktech !

**Comment by schalkms** [comment]
Thanks for your review guys!

**Pull request merged**